lib/transports/__es__/_base.js

```diff
@@ -55,34 +96,26 @@ class Base {
 
     aws4signer(esRequest, this.parent).then(() => {
       this.baseRequest(esRequest, (err, response) => {
-        err = this.handleError(err, response)
         if (err) {
           return callback(err, [])
         }
 
-        const parsed = jsonParser.parse(response.body)
-
-        // Set version based on response
-        if (parsed.version) {
-          const isOpenSearch = parsed.version.distribution === 'opensearch'
-          this.ESFullversion = isOpenSearch
-            ? this.parent.options['force-os-version']
-            : parsed.version.number
-          this.ESversion = this.ESFullversion.split('.')[0]
-
-          // Version check for searchAfter
-          if (this.parent.options.searchAfter && !isOpenSearch) {
-            if (!semver.gte(this.ESFullversion, '7.17.0')) {
-              return callback(new Error('searchAfter requires Elasticsearch 7.17.0 or higher'))
-            }
-          }
+        const parsed = this._parseVersion(response)
+
+        this.parent.emit('debug', `detected ${prefix} distribution = ${parsed.distribution}, ESversion = ${parsed.version}, distribution version = ${parsed.distributionVersion}`)
 
-          this.parent.emit('debug', `detected ${prefix} ${isOpenSearch ? 'opensearch' : 'elasticsearch'} version: ${this.ESversion}`)
-        } else {
-          this.ESversion = 5 // Fallback version
-          this.parent.emit('debug', `unable to detect version, using default: ${this.ESversion} for ${prefix}`)
+        // No need to check elasticsearch-serverless. Those are at least >= v8
+        if (this.parent.options.searchAfter && parsed.distribution === "elasticsearch") {
+          if (!semver.gte(this.ESFullversion, '7.17.0')) {
+            return callback(new Error('searchAfter requires Elasticsearch 7.17.0 or higher'))
+          }
         }
 
+        this.ESFullversion = parsed.version
+        this.ESversion = this.ESFullversion.split('.')[0]
+        this.ESDistribution = parsed.distribution
```

Comment:
i would rather not depend on the distributed name. this can affect future compatibility in the case of name changes. We do no support elastic-serverless. this as not been tested